app/models/direct_file_data.rb

```diff
@@ -186,12 +189,23 @@ def fed_non_taxable_ssb
     fed_ssb - fed_taxable_ssb
   end
 
-  def total_fed_adjustments_identify
-    "wrenches" # TODO
+  def fed_adjustments_claimed
+    adjustments = {
+      fed_educator_expenses: {
+        pdf_label: "ed expenses",
+        xml_label: "Educator Expenses"
+      },
+      fed_student_loan_interest: {
+        pdf_label: "stud loan ded",
+        xml_label: "Student Loan Interest Deduction"
+      }
+    }
+    adjustments.keys.each { |k| adjustments[k][:amount] = df_xml_value(k)&.to_i }
+    adjustments.select { |k, info| info[:amount].present? && info[:amount] > 0 }
   end
```

Comment:
Non-blocking suggestion:

It surprised me to see pdf labels in this file, because it looks like most of the existing code keeps the pdf keys in the ny201_pdf.rb


I think I would have expected those strings to be in the ny201_pdf.rb file but not here. I think of DirectFileData as a wrapper around the direct file return XML that does not need to know anything about the downstream PDFs, and that the Ny201Pdf object is responsible for translating data from direct file or elsewhere into the structure needed for the PDF. I'd prefer to move any pdf-specific logic or labels into the pdf code instead of keeping them in the direct file data.

Similarly, I would expect the It201 do define the XML labels it needs, even if they happen to be the same as the keys we use to pull from the fed return XML.

Reply:
I see what you are saying here, and the appeal was to make it easy to add more types of adjustments in the future (since it is my understanding that there are many). This location may not be the most ideal, but splitting these labels up by xml/pdf will make it harder to track in the future.